### ISSUE TITLE
AC_PROG_OBJCXX in configure.ac and AC_LANG_PROGRAM quoting

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#1165]
**Reported by wking on 31 May 42487759 15:57 UTC**
We've had a bug report on Gentoo trying to build boinc-6.12.42 with automake-1.12 (1).  Automake died with:

```
/usr/share/automake-1.12/am/depend2.am: error: am__fastdepOBJCXX does not appear in AM_CONDITIONAL
/usr/share/automake-1.12/am/depend2.am:   The usual way to define 'am__fastdepOBJCXX' is to add 'AC_PROG_OBJCXX'
/usr/share/automake-1.12/am/depend2.am:   to 'configure.ac' and run 'aclocal' and 'autoconf' again
```
I thought I'd poke my head in upstream and see where the current development efforts.  It looks like you added AC_PROG_OBJCXX (2), but then decided to only use it for Darwin (3).  I think the problem has to do with automake versions, not Mac-vs-Linux.

In a somewhat-related issue, I think your AC_LANG_PROGRAM quoting is off in boinc/m4/libcurl.m4 and boinc/m4/boinc_check_fcgi.m4 (4).  See that section of my patch on the Gentoo Bugzilla for a fix (5).

Apologies for the outdated Gentoo version.  I'll see what I can do about getting a version bump :p.

(1): https://bugs.gentoo.org/show_bug.cgi?id=423081

(2): http://boinc.ssl.berkeley.edu/trac/changeset/25754/boinc

(3): http://boinc.ssl.berkeley.edu/trac/changeset/25779/boinc

(4): http://www.flameeyes.eu/autotools-mythbuster/forwardporting/autoconf.html#idp1113200

(5): https://423081.bugs.gentoo.org/attachment.cgi?id=316089

**Comment by romw** [comment]
**Commented by davea on 7 Aug 42489387 16:26 UTC**
(In [25804]) - Unix build tweaks.  Fixes #1193 (I think)

**Comment by romw** [comment]
**Commented by wking on 16 Sep 42489434 21:14 UTC**
Replying to [davea](comment:2):
> (In [- Unix build tweaks.  Fixes #1193 (I think)

It looks like you fixed the AC_LANG_PROGRAM quoting (thanks!), but I think you're still only using AC_PROG_OBJCXX for Darwin (1), and we still need it on Gentoo as well.  The commit message that made it's inclusion Darwin-specific ([25794](25804]))) doesn't look like it needs to be Darwin-only.  Are there legacy users out there with older automakes that you need to support?

(1): http://boinc.berkeley.edu/trac/browser/boinc/trunk/boinc/configure.ac#L41

**Comment by romw** [comment]
**Commented by davea on 29 Jul 42489499 08:18 UTC**
(In [25805]) - Unix build: remove "if" around AC_PROG_OBJCXX;
    apparently this is needed for Gentoo build.
    You may need a recent automake for this to work.
    Fixes #1193.
